deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/UrlTree.kt

```diff
@@ -52,27 +57,70 @@ open class TreeNode(open val id: String, val type: UByte, open val placeholder:
         return resultByteArray
     }
 
+    /**
+     * Use a UByte for [flag] so that we can easily specify and read different
+     * combinations/permutations of flags.
+     */
+    enum class TransformationType(val flag: UByte) {
+        NoTransformation(2u),
+        /**
+         * String Insertion type means that a string can be added into the uri component at runtime.
+         */
+        StringInsertion(4u),
+        /**
+         * PathSegmentReplacement type represents a path segment which will lookup a dynamic (runtime) provided
+         * replacement value when it is visited.
+         */
+        PathSegmentReplacement(8u)
+    }
+
+    /**
+     * Transformation types:
+     * @return an ASCII encoding as a character for the flag for whichever transformation type
+     * applies.
+     */
+    private fun String.transformationType(): UByte = when {
+        startsWith("%%%") && endsWith("%%%") -> {
```

Comment:
Curious why `%%%` over something simpler like `<` `>`

Reply:
I want it to be really obvious that these are non-standard path segments. It might be better to have differentiated opening/closing characters, though, such as `<<` and `>>`.